### PR TITLE
Dylan/project setup

### PR DESCRIPTION
**Pull request by dstaatz**
Most things are set up and are using error.h

One current problem to note is that navigation.test.c will not successfully complie with error.c as a dependency with `make run`

### CODE REVIEW

**Comment by dstaatz** [comment]
Zach, can you look into why `make run` is failing to complie navigation.test.c?

**Comment by zachdeibert** [comment]
You were just missing a comma

File: software/src/algorithms/navigation.test.c [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+ * navigation.test.c
+ */ 
+
+
+#include <testing.h>
+#include <error.h>
+
+#include "navigation.h"
+
+TEST(navigation compiles, src/algorithms/navigation.c, src/error.c) {
```
Comment (zachdeibert) [inline software/src/algorithms/navigation.test.c:11]: You need to have a comma between each dependent file.  Earlier you had just a space in here.

File: software/src/error.c [hunk 2]
```diff
@@ -1,3 +1,8 @@
+/* 
+ * error.c
+ */
+
+#include <stdint.h>
```
Comment (zachdeibert) [inline software/src/error.c:5]: Was also missing this include, but that wasn't the problem you were having.

**Review by zachdeibert — APPROVED** [review]
Should be good now.

**Pull request merged**